### PR TITLE
New debian validation pipeline (infra)

### PR DESCRIPTION
**Pull request by Hook25**
## Description

This new workflow builds the debian packages when relevant files are changed.

## Resolved issues

We have broken the debian packages twice in the last 2 days

## Documentation

N/A

## Tests

N/A

### CODE REVIEW

File: .github/workflows/deb_validator.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+name: Verify debian packages
+
+on:
+  push:
```
Comment (p-gentili) [inline .github/workflows/deb_validator.yaml:4]: Any situation where we would push w/o PR?

File: .github/workflows/deb_validator.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      - providers/tpm2/**
+  workflow_dispatch:
+
+jobs:
+  deb_validation:
+    strategy:
+      fail-fast: false
+      matrix:
+        # Version of Ubuntu
+        # note: no 16, we dont have 16 debs anymore
```
Comment (p-gentili) [inline .github/workflows/deb_validator.yaml:37]:
```suggestion
        # note: no 16, we don't support debs for 16 anymore
```

**Review by p-gentili — APPROVED** [review]
It would be nice to filter out configurations from the matrix that are not involved. 

Since it only takes 4 minutes to run instead of hours debugging problems with dependencies, we can address this later if it's not trivial.

File: .github/workflows/deb_validator.yaml (hunk 1 above)
Comment (Hook25) [reply .github/workflows/deb_validator.yaml:4]: It is a nice to have to push the branch and work without a pr sometimes

**Comment by Hook25** [comment]
I agree that this is subottimal as is, but it does give a nice barrier to trivial mistakes. I will search for a clean solution to only run what is necessary but the metabox one can't be easily adapted to this so I will do it in a follow up

**Review by p-gentili — APPROVED** [review]
+1

**Pull request merged**